src/jit/morph.cpp

```diff
@@ -7689,12 +7706,36 @@ GenTreePtr Compiler::fgMorphCall(GenTreeCall* call)
                 //
                 if (call->IsImplicitTailCall())
                 {
-                    if (varDsc->lvHasLdAddrOp || varDsc->lvAddrExposed)
+                    if (varDsc->lvHasLdAddrOp)
                     {
                         hasAddrExposedVars = true;
                         break;
                     }
-                    if (varDsc->lvPromoted && varDsc->lvIsParam)
+                    else if (varDsc->lvAddrExposed)
```

Comment:
I you're just seeing github's diff lining up the wrong lines.  I changed

```
if (A || B) {   // note the disjunction here
  X;
}
if (C) {
  Y;
}
```

to

```
if (A) {    // no longer disjunction
  X;
} else if (B) {   // because second clause needs different handling now
  Z;
}
if (C && !D) {
  Y;
}
```

so what was an `if` is still an `if`, not an `else-if` (it's just down on line 7738 now)